src/main/java/com/ldtteam/structurize/blueprints/v1/BlueprintUtil.java

```diff
@@ -109,43 +110,82 @@ public static Blueprint createBlueprint(
         final CompoundTag[] tes = tileEntities.toArray(new CompoundTag[0]);
 
         final List<CompoundTag> entitiesTag = new ArrayList<>();
-
-        List<Entity> entities = new ArrayList<>();
+        final AABB area = new AABB(pos.getX(), pos.getY(), pos.getZ(), pos.getX() + sizeX, pos.getY() + sizeY, pos.getZ() + sizeZ);
         if (saveEntities)
         {
-            entities = world.getEntities(null,
-              new AABB(pos.getX(), pos.getY(), pos.getZ(), pos.getX() + sizeX, pos.getY() + sizeY, pos.getZ() + sizeZ));
-        }
+            for (final Entity entity : world.getEntities(null, area))
+            {
+                if (!entity.getType().canSerialize())
+                {
+                    continue;
+                }
+                entitiesTag.add(entity.serializeNBT());
+            }
 
-        for (final Entity entity : entities)
-        {
-            if (!entity.getType().canSerialize())
+            world.entityManager.permanentStorage.flush(false);
+            for (int x = SectionPos.blockToSectionCoord(pos.getX()); x <= SectionPos.blockToSectionCoord(pos.getX() + sizeX); x++)
             {
-                continue;
+                for (int z = SectionPos.blockToSectionCoord(pos.getZ()); z <= SectionPos.blockToSectionCoord(pos.getZ() + sizeZ); z++)
+                {
+                    if (!world.entityManager.areEntitiesLoaded(new ChunkPos(x, z).toLong()))
```

Comment:
Imho this is not time safe, in means between world.getEntities(null, area) and for { for { blah }} can the chunk turn from unloaded to loaded and data will be missed

Reply:
> Within the same tick?

well, idk how async io is done there

Reply:
> The io result will be picked up and processed in the next tick

Then it's fine, just the nbt copy, since were changing it